src/client/src/ui/spotTile/spotTileReducer.ts

```diff
@@ -34,7 +34,7 @@ const spotTileReducer = (
     case TILE_ACTION_TYPES.SPOT_TILE_SUBSCRIBE:
       return state
     case TILE_ACTION_TYPES.SPOT_PRICES_UPDATE:
-      return { ...state, price: action.payload }
+      return { ...state, price: action.payload, historicPrices: [...state.historicPrices.slice(0), action.payload] }
```

Comment:
what is the intention with `[...state.historicPrices.slice(0), action.payload]` ?
Note that `state.historicPrices.slice(0)` is an array with the same elements as `state.historicPrices`, which is the same effect as spreading the array. Unless I'm missing something:
- if you intend to remove the first element of the array, should use `.slice(1)`
- if you intend to copy all elements and append the payload, just `[...state.historicPrices, action.payload]` will do

Reply:
Oups, thanks for spotting that out, I meant to use slice(1). As new price data is coming in, I am removing the first one. So, I am essentially moving historicPrices by one value to the left.